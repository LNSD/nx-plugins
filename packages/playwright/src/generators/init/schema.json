{
  "$schema": "http://json-schema.org/schema",
  "$id": "NxPlaywrightInit",
  "cli": "nx",
  "title": "Add Playwright Configuration to the workspace",
  "type": "object",
  "properties": {
    "skipPackageJson": {
      "type": "boolean",
      "default": false,
      "description": "Do not add dependencies to `package.json`."
    }
  }
}
