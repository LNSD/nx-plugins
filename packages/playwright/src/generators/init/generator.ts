import {
  addDependenciesToPackageJson,
  convertNxGenerator,
  Tree,
  updateJson,
} from '@nrwl/devkit';

import {
  playwrightVersion,
  playwrightTestVersion,
  nxVersion,
} from '../../utils/playwright-version';
import { InitGeneratorSchema } from './schema';

function updateDependencies(tree: Tree) {
  updateJson(tree, 'package.json', (json) => {
    json.dependencies = json.dependencies || {};
    delete json.dependencies['@lnsd/nx-playwright'];

    return json;
  });

  return addDependenciesToPackageJson(
    tree,
    {},
    {
      '@lnsd/nx-playwright': nxVersion,
      playwright: playwrightVersion,
      '@playwright/test': playwrightTestVersion,
    },
  );
}

export function playwrightInitGenerator(
  host: Tree,
  options: InitGeneratorSchema,
) {
  if (options.skipPackageJson) {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    return () => {};
  }
  return updateDependencies(host);
}

export default playwrightInitGenerator;
export const playwrightInitSchematic = convertNxGenerator(
  playwrightInitGenerator,
);
