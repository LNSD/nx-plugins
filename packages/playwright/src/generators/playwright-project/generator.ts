import {
  addDependenciesToPackageJson,
  addProjectConfiguration,
  readProjectConfiguration,
  convertNxGenerator,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  joinPathFragments,
  names,
  offsetFromRoot,
  toJS,
  Tree,
  updateJson,
  ProjectConfiguration,
} from '@nrwl/devkit';
import { Linter, lintProjectGenerator } from '@nrwl/linter';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { join } from 'path';

import { eslintPluginPlaywrightVersion } from '../../utils/playwright-version';
import { filePathPrefix } from '../../utils/project-name';
import { PlaywrightProjectGeneratorSchema as Schema } from './schema';

export interface PlaywrightProjectSchema extends Schema {
  projectName: string;
  projectRoot: string;
}

function createFiles(tree: Tree, options: PlaywrightProjectSchema) {
  generateFiles(tree, join(__dirname, './files'), options.projectRoot, {
    tmpl: '',
    ...options,
    project: options.project ?? 'Project',
    ext: options.js ? 'js' : 'ts',
    offsetFromRoot: offsetFromRoot(options.projectRoot),
  });

  if (options.js) {
    toJS(tree);
  }
}

function addProject(tree: Tree, options: PlaywrightProjectSchema) {
  const baseE2eProjectConfig: ProjectConfiguration = {
    root: options.projectRoot,
    sourceRoot: joinPathFragments(options.projectRoot, 'src'),
    projectType: 'application',
    targets: {
      e2e: {
        executor: '@lnsd/nx-playwright:test',
        options: {
          playwrightConfig: joinPathFragments(
            options.projectRoot,
            'playwright.config.ts',
          ),
          outputDir: joinPathFragments(
            offsetFromRoot(options.projectRoot),
            'dist',
            'playwright',
            options.projectRoot,
            'test-results',
          ),
        },
        configurations: {
          headed: {
            headed: true,
          },
          debug: {
            debug: true,
          },
        },
      },
    },
    tags: [],
    implicitDependencies: options.project ? [options.project] : undefined,
  };
  let e2eProjectConfig: ProjectConfiguration;

  if (!options.baseUrl && !options.project) {
    throw new Error(`Either "project" or "baseUrl" should be specified.`);
  } else if (options.baseUrl) {
    e2eProjectConfig = {
      ...baseE2eProjectConfig,
      targets: {
        ...baseE2eProjectConfig.targets,
        e2e: {
          ...baseE2eProjectConfig.targets.e2e,
          options: {
            ...baseE2eProjectConfig.targets.e2e.options,
            baseUrl: options.baseUrl,
          },
        },
      },
    };
  } else if (options.project) {
    const project = readProjectConfiguration(tree, options.project);

    let devServerTarget = `${options.project}:serve`;
    if (project.targets.serve?.defaultConfiguration) {
      devServerTarget = `${options.project}:serve:${project.targets.serve.defaultConfiguration}`;
    }

    e2eProjectConfig = {
      ...baseE2eProjectConfig,
      targets: {
        ...baseE2eProjectConfig.targets,
        e2e: {
          ...baseE2eProjectConfig.targets.e2e,
          options: {
            ...baseE2eProjectConfig.targets.e2e.options,
            devServerTarget,
          },
          configurations: {
            ...baseE2eProjectConfig.targets.e2e.configurations,
            production: {
              devServerTarget: `${options.project}:serve:production`,
            },
          },
        },
      },
    };
  }

  addProjectConfiguration(
    tree,
    options.projectName,
    e2eProjectConfig,
    options.standaloneConfig,
  );
}

export async function addLinter(host: Tree, options: PlaywrightProjectSchema) {
  if (options.linter === Linter.None) {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    return () => {};
  }

  const installTask = await lintProjectGenerator(host, {
    project: options.projectName,
    linter: options.linter,
    skipFormat: true,
    tsConfigPaths: [joinPathFragments(options.projectRoot, 'tsconfig.json')],
    eslintFilePatterns: [
      `${options.projectRoot}/**/*.${options.js ? 'js' : '{js,ts}'}`,
    ],
    setParserOptionsProject: options.setParserOptionsProject,
    skipPackageJson: options.skipPackageJson,
  });

  if (!options.linter || options.linter !== Linter.EsLint) {
    return installTask;
  }

  let eslintInstallTask;
  if (options.skipPackageJson) {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    eslintInstallTask = () => {};
  } else {
    eslintInstallTask = addDependenciesToPackageJson(
      host,
      {},
      {
        'eslint-plugin-playwright': eslintPluginPlaywrightVersion,
      },
    );
  }

  updateJson(host, join(options.projectRoot, '.eslintrc.json'), (json) => {
    json.extends = ['plugin:playwright/playwright-test', ...json.extends];
    json.overrides = [
      /**
       * In order to ensure maximum efficiency when typescript-eslint generates TypeScript Programs
       * behind the scenes during lint runs, we need to make sure the project is configured to use its
       * own specific tsconfigs, and not fall back to the ones in the root of the workspace.
       */
      {
        files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
        /**
         * NOTE: We no longer set parserOptions.project by default when creating new projects.
         *
         * We have observed that users rarely add rules requiring type-checking to their Nx workspaces, and therefore
         * do not actually need the capabilities which parserOptions.project provides. When specifying parserOptions.project,
         * typescript-eslint needs to create full TypeScript Programs for you. When omitting it, it can perform a simple
         * parse (and AST transformation) of the source files it encounters during a lint run, which is much faster and much
         * less memory intensive.
         *
         * In the rare case that users attempt to add rules requiring type-checking to their setup later on (and haven't set
         * parserOptions.project), the executor will attempt to look for the particular error typescript-eslint gives you
         * and provide feedback to the user.
         */
        parserOptions: options.setParserOptionsProject
          ? { project: `${options.projectRoot}/tsconfig.*?.json` }
          : undefined,
        /**
         * Having an empty rules object present makes it more obvious to the user where they would
         * extend things from if they needed to
         */
        rules: {},
      },
    ];

    return json;
  });

  return runTasksInSerial(installTask, eslintInstallTask);
}

export async function playwrightProjectGenerator(host: Tree, schema: Schema) {
  const options = normalizeOptions(host, schema);
  createFiles(host, options);
  addProject(host, options);
  const linterInstallTask = await addLinter(host, options);
  if (!options.skipFormat) {
    await formatFiles(host);
  }
  return linterInstallTask;
}

function normalizeOptions(
  host: Tree,
  options: Schema,
): PlaywrightProjectSchema {
  const { appsDir } = getWorkspaceLayout(host);
  const projectName = filePathPrefix(
    options.directory ? `${options.directory}-${options.name}` : options.name,
  );

  let projectRoot: string;
  if (options.directory) {
    projectRoot = joinPathFragments(
      appsDir,
      names(options.directory).fileName,
      options.name,
    );
  } else {
    projectRoot = joinPathFragments(appsDir, options.name);
  }

  options.linter = options.linter ?? Linter.EsLint;
  return {
    ...options,
    projectName,
    projectRoot,
  };
}

export default playwrightProjectGenerator;
export const playwrightProjectSchematic = convertNxGenerator(
  playwrightProjectGenerator,
);
