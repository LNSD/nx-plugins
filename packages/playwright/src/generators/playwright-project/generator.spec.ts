import {
  readJson,
  addProjectConfiguration,
  readProjectConfiguration,
  updateProjectConfiguration,
  Tree,
  WorkspaceJsonConfiguration,
} from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Linter } from '@nrwl/linter';

import { playwrightProjectGenerator } from './generator';
import { PlaywrightProjectGeneratorSchema as Schema } from './schema';

describe('generator:playwright-project', () => {
  let tree: Tree;
  const defaultOptions: Omit<Schema, 'name' | 'project'> = {
    linter: Linter.EsLint,
    standaloneConfig: false,
  };

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();

    addProjectConfiguration(tree, 'my-app', {
      root: 'my-app',
      targets: {
        serve: {
          executor: 'serve-executor',
          options: {},
          configurations: {
            production: {},
          },
        },
      },
    });

    addProjectConfiguration(tree, 'my-dir-my-app', {
      root: 'my-dir/my-app',
      targets: {
        serve: {
          executor: 'serve-executor',
          options: {},
          configurations: {
            production: {},
          },
        },
      },
    });
  });

  describe('Playwright Project', () => {
    it('should generate files', async () => {
      await playwrightProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });

      expect(tree.exists('apps/my-app-e2e/playwright.config.ts')).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/app.spec.ts')).toBeTruthy();
      expect(tree.exists('apps/my-app-e2e/src/support/app.po.ts')).toBeTruthy();
    });

    it('should add update `workspace.json` file', async () => {
      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.TsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect.lint).toEqual({
        builder: '@angular-devkit/build-angular:tslint',
        options: {
          tsConfig: ['apps/my-app-e2e/tsconfig.json'],
          exclude: ['**/node_modules/**', '!apps/my-app-e2e/**/*'],
        },
      });
      expect(project.architect.e2e).toEqual({
        builder: '@lnsd/nx-playwright:test',
        options: {
          playwrightConfig: 'apps/my-app-e2e/playwright.config.ts',
          outputDir: '../../dist/playwright/apps/my-app-e2e/test-results',
          devServerTarget: 'my-app:serve',
        },
        configurations: {
          production: {
            devServerTarget: 'my-app:serve:production',
          },
          debug: {
            debug: true,
          },
          headed: {
            headed: true,
          },
        },
      });
    });

    it('should add update `workspace.json` file (baseUrl)', async () => {
      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        baseUrl: 'http://localhost:3000',
        linter: Linter.TsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect.lint).toEqual({
        builder: '@angular-devkit/build-angular:tslint',
        options: {
          tsConfig: ['apps/my-app-e2e/tsconfig.json'],
          exclude: ['**/node_modules/**', '!apps/my-app-e2e/**/*'],
        },
      });
      expect(project.architect.e2e).toEqual({
        builder: '@lnsd/nx-playwright:test',
        options: {
          playwrightConfig: 'apps/my-app-e2e/playwright.config.ts',
          outputDir: '../../dist/playwright/apps/my-app-e2e/test-results',
          baseUrl: 'http://localhost:3000',
        },
        configurations: {
          debug: {
            debug: true,
          },
          headed: {
            headed: true,
          },
        },
      });
    });

    it('should add update `workspace.json` file for a project with a defaultConfiguration', async () => {
      const originalProject = readProjectConfiguration(tree, 'my-app');
      originalProject.targets.serve.defaultConfiguration = 'development';
      originalProject.targets.serve.configurations.development = {};
      updateProjectConfiguration(tree, 'my-app', originalProject);

      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.TsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.root).toEqual('apps/my-app-e2e');

      expect(project.architect.lint).toEqual({
        builder: '@angular-devkit/build-angular:tslint',
        options: {
          tsConfig: ['apps/my-app-e2e/tsconfig.json'],
          exclude: ['**/node_modules/**', '!apps/my-app-e2e/**/*'],
        },
      });
      expect(project.architect.e2e).toEqual({
        builder: '@lnsd/nx-playwright:test',
        options: {
          playwrightConfig: 'apps/my-app-e2e/playwright.config.ts',
          outputDir: '../../dist/playwright/apps/my-app-e2e/test-results',
          devServerTarget: 'my-app:serve:development',
        },
        configurations: {
          production: {
            devServerTarget: 'my-app:serve:production',
          },
          debug: {
            debug: true,
          },
          headed: {
            headed: true,
          },
        },
      });
    });

    it('should add update `workspace.json` file properly when eslint is passed', async () => {
      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.architect.lint).toEqual({
        builder: '@nrwl/linter:eslint',
        outputs: ['{options.outputFile}'],
        options: {
          lintFilePatterns: ['apps/my-app-e2e/**/*.{js,ts}'],
        },
      });
    });

    it('should not add lint target when "none" is passed', async () => {
      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.None,
        standaloneConfig: false,
      });
      const workspaceJson = readJson(tree, 'workspace.json');
      const project = workspaceJson.projects['my-app-e2e'];

      expect(project.architect.lint).toBeUndefined();
    });

    it('should update tags and implicit dependencies', async () => {
      await playwrightProjectGenerator(tree, {
        name: 'my-app-e2e',
        project: 'my-app',
        linter: Linter.EsLint,
        standaloneConfig: false,
      });

      const project = readProjectConfiguration(tree, 'my-app-e2e');
      expect(project.tags).toEqual([]);
      expect(project.implicitDependencies).toEqual(['my-app']);
    });

    // TODO: Review configuration parsing
    it.skip('should set right path names in `playwright.config.ts`', async () => {
      await playwrightProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });
      const playwrightJson = readJson(
        tree,
        'apps/my-app-e2e/playwright.config.ts',
      );

      expect(playwrightJson).toEqual({
        fileServerFolder: '.',
        fixturesFolder: './src/fixtures',
        integrationFolder: './src/integration',
        modifyObstructiveCode: false,
        pluginsFile: './src/plugins/index',
        supportFile: './src/support/index.ts',
        video: true,
        videosFolder: '../../dist/playwright/apps/my-app-e2e/videos',
        screenshotsFolder: '../../dist/playwright/apps/my-app-e2e/screenshots',
        chromeWebSecurity: false,
      });
    });

    it('should set right path names in `tsconfig.e2e.json`', async () => {
      await playwrightProjectGenerator(tree, {
        ...defaultOptions,
        name: 'my-app-e2e',
        project: 'my-app',
      });
      const tsconfigJson = readJson(tree, 'apps/my-app-e2e/tsconfig.json');
      expect(tsconfigJson).toMatchSnapshot();
    });

    describe('nested', () => {
      it('should update workspace.json', async () => {
        await playwrightProjectGenerator(tree, {
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
          linter: Linter.TsLint,
          standaloneConfig: false,
        });
        const projectConfig = readJson(tree, 'workspace.json').projects[
          'my-dir-my-app-e2e'
        ];

        expect(projectConfig).toBeDefined();
        expect(projectConfig.architect.lint).toEqual({
          builder: '@angular-devkit/build-angular:tslint',
          options: {
            tsConfig: ['apps/my-dir/my-app-e2e/tsconfig.json'],
            exclude: ['**/node_modules/**', '!apps/my-dir/my-app-e2e/**/*'],
          },
        });

        expect(projectConfig.architect.e2e).toEqual({
          builder: '@lnsd/nx-playwright:test',
          options: {
            playwrightConfig: 'apps/my-dir/my-app-e2e/playwright.config.ts',
            outputDir:
              '../../../dist/playwright/apps/my-dir/my-app-e2e/test-results',
            devServerTarget: 'my-dir-my-app:serve',
          },
          configurations: {
            production: {
              devServerTarget: 'my-dir-my-app:serve:production',
            },
            debug: {
              debug: true,
            },
            headed: {
              headed: true,
            },
          },
        });
      });

      // TODO: Review configuration parsing
      it.skip('should set right path names in `playwright.config.ts`', async () => {
        await playwrightProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
        });
        const playwrightJson = readJson(
          tree,
          'apps/my-dir/my-app-e2e/playwright.config.ts',
        );

        expect(playwrightJson).toEqual({
          fileServerFolder: '.',
          fixturesFolder: './src/fixtures',
          integrationFolder: './src/integration',
          modifyObstructiveCode: false,
          pluginsFile: './src/plugins/index',
          supportFile: './src/support/index.ts',
          video: true,
          videosFolder:
            '../../../dist/playwright/apps/my-dir/my-app-e2e/videos',
          screenshotsFolder:
            '../../../dist/playwright/apps/my-dir/my-app-e2e/screenshots',
          chromeWebSecurity: false,
        });
      });

      it('should set right path names in `tsconfig.e2e.json`', async () => {
        await playwrightProjectGenerator(tree, {
          ...defaultOptions,
          name: 'my-app-e2e',
          project: 'my-dir-my-app',
          directory: 'my-dir',
        });
        const tsconfigJson = readJson(
          tree,
          'apps/my-dir/my-app-e2e/tsconfig.json',
        );

        expect(tsconfigJson).toMatchSnapshot();
      });
    });

    describe('--project', () => {
      describe('none', () => {
        it('should not add any implicit dependencies', async () => {
          await playwrightProjectGenerator(tree, {
            ...defaultOptions,
            name: 'my-app-e2e',
            baseUrl: 'http://localhost:7788',
          });

          const workspaceJson = readJson<WorkspaceJsonConfiguration>(
            tree,
            'workspace.json',
          );
          const projectConfig = workspaceJson.projects['my-app-e2e'];
          expect(projectConfig.implicitDependencies).not.toBeDefined();
          expect(projectConfig.tags).toEqual([]);
        });
      });
    });

    describe('--linter', () => {
      describe('eslint', () => {
        it('should add eslint-plugin-playwright', async () => {
          await playwrightProjectGenerator(tree, {
            name: 'my-app-e2e',
            project: 'my-app',
            linter: Linter.EsLint,
            standaloneConfig: false,
          });
          const packageJson = readJson(tree, 'package.json');
          expect(
            packageJson.devDependencies['eslint-plugin-playwright'],
          ).toBeTruthy();

          const eslintrcJson = readJson(tree, 'apps/my-app-e2e/.eslintrc.json');
          expect(eslintrcJson).toMatchInlineSnapshot(`
            Object {
              "extends": Array [
                "plugin:playwright/playwright-test",
                "../../.eslintrc.json",
              ],
              "ignorePatterns": Array [
                "!**/*",
              ],
              "overrides": Array [
                Object {
                  "files": Array [
                    "*.ts",
                    "*.tsx",
                    "*.js",
                    "*.jsx",
                  ],
                  "rules": Object {},
                },
              ],
            }
          `);
        });
      });
    });
    it('should generate in the correct folder', async () => {
      await playwrightProjectGenerator(tree, {
        ...defaultOptions,
        name: 'other-e2e',
        project: 'my-app',
        directory: 'one/two',
      });
      const workspace = readJson(tree, 'workspace.json');
      expect(workspace.projects['one-two-other-e2e']).toBeDefined();
      [
        'apps/one/two/other-e2e/playwright.config.ts',
        'apps/one/two/other-e2e/src/app.spec.ts',
      ].forEach((path) => expect(tree.exists(path)).toBeTruthy());
    });

    describe('project with directory in its name', () => {
      beforeEach(async () => {
        await playwrightProjectGenerator(tree, {
          name: 'my-dir/my-app-e2e',
          project: 'my-dir-my-app',
          linter: Linter.TsLint,
          standaloneConfig: false,
        });
      });

      it('should update workspace.json', async () => {
        const projectConfig = readJson(tree, 'workspace.json').projects[
          'my-dir-my-app-e2e'
        ];

        expect(projectConfig).toBeDefined();
        expect(projectConfig.architect.lint).toEqual({
          builder: '@angular-devkit/build-angular:tslint',
          options: {
            tsConfig: ['apps/my-dir/my-app-e2e/tsconfig.json'],
            exclude: ['**/node_modules/**', '!apps/my-dir/my-app-e2e/**/*'],
          },
        });

        expect(projectConfig.architect.e2e).toEqual({
          builder: '@lnsd/nx-playwright:test',
          options: {
            playwrightConfig: 'apps/my-dir/my-app-e2e/playwright.config.ts',
            outputDir:
              '../../../dist/playwright/apps/my-dir/my-app-e2e/test-results',
            devServerTarget: 'my-dir-my-app:serve',
          },
          configurations: {
            production: {
              devServerTarget: 'my-dir-my-app:serve:production',
            },
            debug: {
              debug: true,
            },
            headed: {
              headed: true,
            },
          },
        });
      });

      it('should update nx.json', async () => {
        const project = readProjectConfiguration(tree, 'my-dir-my-app-e2e');
        expect(project.tags).toEqual([]);
        expect(project.implicitDependencies).toEqual(['my-dir-my-app']);
      });

      // TODO: Review configuration parsing
      it.skip('should set right path names in `playwright.json`', async () => {
        const playwrightJson = readJson(
          tree,
          'apps/my-dir/my-app-e2e/playwright.json',
        );

        expect(playwrightJson).toEqual({
          fileServerFolder: '.',
          fixturesFolder: './src/fixtures',
          integrationFolder: './src/integration',
          modifyObstructiveCode: false,
          pluginsFile: './src/plugins/index',
          supportFile: './src/support/index.ts',
          video: true,
          videosFolder:
            '../../../dist/playwright/apps/my-dir/my-app-e2e/videos',
          screenshotsFolder:
            '../../../dist/playwright/apps/my-dir/my-app-e2e/screenshots',
          chromeWebSecurity: false,
        });
      });
    });
  });
});
