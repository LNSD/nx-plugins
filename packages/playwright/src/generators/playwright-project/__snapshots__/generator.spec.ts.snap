// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator:playwright-project Playwright Project nested should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "node",
    ],
  },
  "extends": "../../../tsconfig.base.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
  ],
}
`;

exports[`generator:playwright-project Playwright Project should set right path names in \`tsconfig.e2e.json\` 1`] = `
Object {
  "compilerOptions": Object {
    "allowJs": true,
    "outDir": "../../dist/out-tsc",
    "sourceMap": false,
    "types": Array [
      "node",
    ],
  },
  "extends": "../../tsconfig.base.json",
  "include": Array [
    "src/**/*.ts",
    "src/**/*.js",
  ],
}
`;
